testsuite/certificates/__init__.py

```diff
@@ -0,0 +1,158 @@
+                     hosts: Optional[List[str]] = None) -> UnsignedKey:
+        """Generates unsigned key"""
+        data: Dict[str, Any] = {
+            "CN": common_name
+        }
+        if names:
+            data["names"] = names
+        if hosts:
+            data["hosts"] = hosts
+
+        result = self._execute_command("genkey", "-", stdin=json.dumps(data))
+        return UnsignedKey(key=result["key"], csr=result["csr"])
+
+    def sign_intermediate_ca(self, key: UnsignedKey, certificate_authority: Certificate) -> Certificate:
+        """Signs intermediate ca"""
+        args = [
+            "-ca=env:CA",
+            "-ca-key=env:KEY",
+            f"-config={resources.files('testsuite.resources.tls').joinpath('intermediate_config.json')}"
+        ]
+        result = self._execute_command("sign", *args, "-", stdin=key.csr, env={
+            "CA": certificate_authority.certificate,
+            "KEY": certificate_authority.key})
+        return Certificate(key=key.key, certificate=result["cert"])
+
+    def sign(self, key: UnsignedKey, certificate_authority: Certificate) -> Certificate:
+        """Signs unsigned key"""
+        result = self._execute_command("sign", "-ca=env:CA", "-ca-key=env:KEY", "-", stdin=key.csr, env={
+            "CA": certificate_authority.certificate,
+            "KEY": certificate_authority.key})
+        return Certificate(key=key.key, certificate=result["cert"])
+
+    def generate_ca(self, common_name: str,
+                    names: List[Dict[str, str]],
+                    hosts: List[str]) -> Tuple[Certificate, UnsignedKey]:
+        """Generates CA"""
+        data = {
+            "CN": common_name,
+            "names": names,
+            "hosts": hosts,
+            "key": {
+                "algo": "rsa",
+                "size": 4096
+            },
+        }
+
+        result = self._execute_command("genkey", "-initca", "-", stdin=json.dumps(data))
+        key = UnsignedKey(key=result["key"], csr=result["csr"])
+        cert = Certificate(key=result["key"], certificate=result["cert"])
+        return cert, key
+
+    def create_ca(self,
+                  label: str,
+                  hosts: List[str],
+                  names: Optional[List[Dict[str, str]]] = None,
+                  certificate_authority: Optional[Certificate] = None) -> Certificate:
+        """Create a new certificate authority, if ca is specified it will create an intermediate ca.
+        Args:
+            :param certificate_authority:
+            :param names: dict of all names
+            :param hosts: list of hosts
+            :param label: a identifier to the certificate and key.
+        """
+        names = names or self.DEFAULT_NAMES
+        certificate, key = self.generate_ca(label, names, hosts)
+        if certificate_authority:
+            certificate = self.sign_intermediate_ca(key, certificate_authority)
+        return certificate
+
+    def create(self,
```

Comment:
This is a similar method to `generate_key`, but this one returns a signed cert? If yes, I'd say keep the naming similar, like in the case of CA methods and move these methods together.